### ISSUE TITLE
Adding poetry-release to poetry 1.6.1 breaks poetry.

### ISSUE THREAD

**Issue by jflipts** [topenkoff/poetry-release#6]
When I add poetry-release to poetry 1.6.1 with `poetry self add poetry-release` (^0.3.1) this works but running any poetry command after that results in `No module named 'poetry.core.semver'`. 

Before adding poetry-release this is the output of `poetry env info`
```
Virtualenv
Python:         3.10.9
Implementation: CPython
Path:           NA
Executable:     NA

System
Platform:   linux
OS:         posix
Python:     3.10.9
Path:       /home/jflipts/miniconda3
Executable: /home/jflipts/miniconda3/bin/python3.10
```

When I uninstall poetry and install poetry 1.5.1 and add poetry-release 0.3.1 everything works back as expected.

**Comment by pcrumley** [comment]
Yeah the issue is that poetry-core deprecated that code: https://github.com/python-poetry/poetry-core/releases/tag/1.7.0 , it should be using the replacement code at: `poetry.core.constraints.version`. I'd love to know if the maintainer of this lib has plans to update the poetry release to support the new location for the semver code, or if they would be willing to review PRs that could fix this issue.

Thanks
